src/diffusers/quantizers/bitsandbytes/bnb_quantizer.py

```diff
@@ -370,7 +385,9 @@ def validate_environment(self, *args, **kwargs):
             device_map_without_no_convert = {
                 key: device_map[key] for key in device_map.keys() if key not in self.modules_to_not_convert
             }
-            if "cpu" in device_map_without_no_convert.values() or "disk" in device_map_without_no_convert.values():
+            if set(device_map.values()) == {"cpu"} and bnb_multibackend_is_enabled:
+                pass
+            elif "cpu" in device_map_without_no_convert.values() or "disk" in device_map_without_no_convert.values():
```

Comment:
The common piece of code between the two utilities could be clubbed into a small function and reused? 

Previously we didn't do because it was relatively small and was better off in-line.